src/lib/OS/SimOS.cc

```diff
@@ -28,19 +28,24 @@ SimOS::SimOS(std::shared_ptr<simeng::memory::Mem> mem,
 // The Private constructor
 SimOS::SimOS(std::shared_ptr<simeng::memory::Mem> mem)
     : memory_(mem),
-      syscallHandler_(std::make_shared<SyscallHandler>(this)),
+      syscallHandler_(std::make_shared<SyscallHandler>(this, mem)),
```

Comment:
This is the private constructor, so would it not work given `this` will not change after this constructor has concluded?

Reply:
no initialiser lists don't work when we have a delegating constructor. The delegating constructor can have the initialiser list.I think the rebase will flag it anyway

Reply:
Also, I'm not sure if `this` should be passed to any member in the initialiser list (Given we will have a delegating constructor), because of the initialisation order so until we know for sure could it please be moved to the constructor.

Reply:
But this is the delegating constructor, so the initialiser list will work. Also I'm sure that something in one of the Jenkins pipelines would have picked it up were it an issue / the compiler